src/shared/utils/index.js

```diff
@@ -202,7 +204,7 @@ const setChartMargin = (
   let legendLabelContainerWidth
   let legendItemWidth
 
-  if (showLegend) {
+  if (showLegend && !disableLegend) {
```

Comment:
Minor but you could add the check for `disable legend` inside where `showLegend` is built in line 194 and leave the rest the same:
> let showLegend = !disableLegend
&& width >= WIDTH_BREAKPOINT_3
                   && columnLegendHeight <= height - 2 * BUFFER

Reply:
_blocking:_ It also removes the need for the change below. As much as possible, consolidate if/else toggles "upstream". Simpler to maintain.